boot/assets.go

```diff
@@ -292,21 +290,16 @@ func (o *TrustedAssetsInstallObserver) ObserveExistingTrustedRecoveryAssets(reco
 		// assets do not need tracking, nothing to do
 		return nil
 	}
-	bl, err := bootloader.Find(recoveryRootDir, &bootloader.Options{
+	bl, trustedAssets, _, err := findMaybeTrustedBootloaderAndAssets(recoveryRootDir, &bootloader.Options{
```

Comment:
if we moved this to *ForModel already, we could also apply the logic that if all the assets are empty and useEncryption is false we could return ErrObserverNotApplicable